src/compiler/crystal/macros.cr

```diff
@@ -634,6 +634,10 @@ module Crystal::Macros
     def reduce(&block) : ASTNode
     end
 
+    # Similar to `Enumerable#reduce`
+    def reduce(memo : ASTNode, &block) : ASTNode
```

Comment:
```suggestion
    def reduce(memo : ASTNode, &) : ASTNode
```

The new syntax for non-captured blocks can now be used.

Reply:
The thing is, this makes it now different from the existing `#reduce` signature. It's probably better to stay with `&block` for both of them.

We should have a global run to replace all uncaptured `&block` arguments with `&`. But it's safer to do this after 0.32.0 is released so it still compiles with pre 0.31.0 compiler. (Not super important, but there is no hurry either)

Reply:
ok guys, i changed it there and back again, no problem .. Please just make some consensus.